\documentclass[a0,portrait]{a0poster}

\usepackage{multicol}
\columnsep=100pt % Amount of white space between the columns
\columnseprule=3pt % Thickness of the black line between the columns

\usepackage[svgnames]{xcolor}

\usepackage{graphicx} % Required for including images
\graphicspath{{img/}} % Location of the graphics files
\usepackage{booktabs} % Top and bottom rules for table
\usepackage[font=small,
            labelfont=bf]{caption}
\usepackage{amsfonts, amsmath, amsthm, amssymb}
\usepackage{wrapfig}
\usepackage[maxnames=4,
            style=numeric,
            backend=biber,
            alldates=year,
            doi=false,
            isbn=false,
            eprint=false,
            url=false]{biblatex}
\bibliography{kwip}

\begin{document}

\begin{minipage}[b]{0.6\linewidth}
\Huge \textbf{\texttt{kWIP}: The $k$-mer Weighted Inner Product}\\[0.9cm]
\Large \textbf{K Murray$^1$, C Webers$^2$, C S Ong$^2$, S For\^et$^3$,
               J Borevitz$^1$, N Warthmann$^1$}\\[0.5cm]
\large $^1$: ARC CoE in Plant Energy Biology, ANU, Canberra\\[0.2cm]
\large $^2$: NICTA, Canberra\\[0.4cm]
\large $^3$: Research School of Biology, ANU, Canberra\\[0.4cm]
\large \texttt{kevin.murray@anu.edu.au} \hspace{1cm} Poster URL: \large
  \url{http://git.io/vcxYF}\\
\end{minipage}
\begin{minipage}[b]{0.4\linewidth}
  \hspace{5cm}
\includegraphics[width=20cm]{logo.png}\\
\end{minipage}

\vspace{1cm} % A bit of extra whitespace between the header and poster content

%------------------------------------------------------------------------------
\begin{multicols}{2}

%------------------------------------------------------------------------------
%	ABSTRACT
%------------------------------------------------------------------------------

\begin{abstract}
\vspace{5mm}

Modern techniques in population genomics generate unprecedented quantities of
data within which complex genetic histories reside. The scale and complexity of
these data require the development of new approaches to the analysis of genetic
data. We present the $k$-mer Weighted Inner Product, a \textit{de novo},
alignment free measure of genetic similarity between samples in a population.
\texttt{kWIP}, is an efficient tool implementing this metric that can determine
the genetic relatedness between samples without alignment or assembly. We show
\texttt{kWIP} can reconstruct the true relatedness between samples directly
from sequencing reads generated with various modern sequencing platforms, as
well as from simulated data.

\texttt{kWIP} works by decomposing sequencing reads to short $k$-mers, hashing
these $k$-mers using a constant-memory data structure, and performing pairwise
distance calculation between these sample $k$-mer hashes. The power of
\texttt{kWIP} comes from the weighting applied across different hash values,
which decreases the effect of erroneous, rare or over-abundant $k$-mers while
focusing on k-mers which give the most insight into the similarity of samples.
\texttt{kWIP} is free, open source software implemented in C++ and released
under the GNU LGPL v3.

\end{abstract}

%------------------------------------------------------------------------------
%	INTRODUCTION
%------------------------------------------------------------------------------

\section*{Introduction}

Modern population genomics requires sequencing many thousands of samples. To
distill knowelege from this data requires analysis by sequence comparison. To
compare such datasets, algorithmic improvement is required. Alignment-free
sequence comparison is a powerful field of algorithms which overcome some
shortcomings of sequence alignment. However, few alignnment free algorthims can
process raw data from modern sequencing platforms, which sequence genomes as
millions of short fragments. kWIP extends alignment-free sequence comparison
algorithms to accept sequencing data directly.

%------------------------------------------------------------------------------
%	MATERIALS AND METHODS
%------------------------------------------------------------------------------

\section*{Algorithms}

\texttt{kWIP} works by decomposing sequencing reads to short $k$-mers, hashing
these $k$-mers using a constant-memory data structure, and performing pairwise
distance calculation between these sample $k$-mer hashes. One can calculate
the inner product between hashes as a similarity measure. However, this treats
all $k$-mers as of equal importance and accuracy. Therefore, \texttt{kWIP}
applies a weight to each $k$-mer to reduce the contribution of technical noise
to the overall signal, and focus on $k$-mers which provide maxmial information
about relatedness within a population.

%------------------------------------------------

\subsection*{Hashing}

Sequence reads are decomposed into $k$-mers and counted in a proababilistic
data structure (Hash). This hashing is performed using the \texttt{khmer} C++
library \cite{crusoe_khmer_2015}.

\begin{center}
  \vspace{1cm}
  \includegraphics[width=20cm]{hashing.png}
  \vspace{1cm}
\end{center}

\subsection*{Entropy vector weighting}

To calculate the weighting applied to each $k$-mer, we first calculate the
frequency of occurance of the $k$-mer in the population. This is simply the
proportion of samples with non-zero counts of a given $k$-mer.

\begin{center}
  \vspace{1cm}
  \includegraphics[width=25cm]{freq_vector.png}
  \vspace{1cm}
\end{center}

Shannon entropy is calculated per (\ref{eqn:shannon}).

\begin{equation}
  H = \sum_{i} P(x_i) -log_2(P(x_i)
\label{eqn:shannon}
\end{equation}

\subsection*{Inner Product Calculation}

Inner products are calcualted pairwise between all samples, and represents the
cosine simiarity in inner product space of the two hash vectors. The inner
product between two hashes alone is calculated as (\ref{eqn:ip}). The weighted
inner product calculation is calculated per (\ref{eqn:wip}).

\begin{equation}
  \langle A, B \rangle = \sum_{i} A_i \cdot B_i
\label{eqn:ip}
\end{equation}

\begin{equation}
  \langle A, B \rangle = \sum_{i} A_i \cdot B_i \cdot H_i
\label{eqn:wip}
\end{equation}

\subsection*{Implementation}

\texttt{kWIP} is impemented in C++11, utilising the khmer C++ library. Weighted
and unweighted inner products have been implemented. \texttt{kWIP} uses OpenMP
to parallelise distance matrix calculation in a threadsafe manner. We use
a shared-memory LRU cache of hashes to reduce computation time and memory
usage.

%------------------------------------------------------------------------------
%	RESULTS
%------------------------------------------------------------------------------

\section*{Experimental Validation}

We present an initial experimental validation of \texttt{kWIP} and show
increased performance of the weighted inner product metric compared to the
unweighted metric.

\subsection*{Methods}

Random genomes with equal base composition, and genetic variation from these
genomes, were simulated using the mason genome simulation suite (parameters
below) and \texttt{seq-gen}. Reads were hashed using \texttt{khmer} version
2.0, and \texttt{kwip} was run with default parameters in both weighted and
unweighed mode.

Specifically, we simulate a 1MBp genome and four populations derived from this
genome (see tree below). We then simulated three samples from each population,
at four coverages (1x, 2x, 5x, 10x).

\subsection*{Simulation Results}


\subsection*{Computational Efficiency}

\texttt{kWIP} can calculate pairwise relatedness 

%------------------------------------------------------------------------------
%	CONCLUSIONS
%------------------------------------------------------------------------------

\color{SaddleBrown}

\section*{Conclusions}

\begin{itemize}
  \item \texttt{kWIP} is a fast tool to determine approximate genetic
    relatedness in an entirely \textit{de novo} and alignnent free manner.
  \item Hashing $k$-mers into probabilistic data structures enables the rapid
    estimation of population hashes.
\end{itemize}

\color{Black}

%------------------------------------------------------------------------------
%	FORTHCOMING RESEARCH
%------------------------------------------------------------------------------

\section*{Forthcoming Research}

A paper describing \texttt{kWIP} in more detail is in preparation. We plan to
deploy \texttt{kWIP} for use as a quality assurance measure across several
large-scale plant population genome sequencing projects. We plan to implement
MPI-based paralellism, and to extend kwip to full counting Bloom filters or
counting de Bruijin graphs.

%------------------------------------------------------------------------------
%	REFERENCES
%------------------------------------------------------------------------------
\printbibliography

%------------------------------------------------------------------------------
%	ACKNOWLEDGEMENTS
%------------------------------------------------------------------------------

\section*{Acknowledgements}

We thank Conrad Burden, Terry Neeman, Ben Kahler, Gavin Huttley and Cameron
Jack for comments or advice on algorithms and experiments presented here.

%------------------------------------------------------------------------------

\end{multicols}
\end{document}
