\documentclass[a0,portrait]{a0poster}

\usepackage{multicol}
\columnsep=100pt % Amount of white space between the columns
\columnseprule=3pt % Thickness of the black line between the columns

\usepackage[svgnames]{xcolor}

\usepackage{graphicx} % Required for including images
\graphicspath{{img/}} % Location of the graphics files
\usepackage{booktabs} % Top and bottom rules for table
\usepackage[font=small,
            labelfont=bf]{caption}
\usepackage{amsfonts, amsmath, amsthm, amssymb}
\usepackage{wrapfig}
\usepackage{fancybox}
\usepackage[maxnames=4,
            style=numeric,
            backend=biber,
            alldates=year,
            doi=false,
            isbn=false,
            eprint=false,
            url=false]{biblatex}
\bibliography{kwip}
\renewcommand{\bibfont}{\normalfont\small}

\begin{document}

\begin{minipage}[b]{0.6\linewidth}
\Huge \textbf{\texttt{kWIP}: The $k$-mer Weighted Inner Product}\\[0.9cm]
\Large \textbf{K Murray$^{1,3}$, C Webers$^2$, C S Ong$^2$, J Borevitz$^{1,3}$, N Warthmann$^{1,3}$}\\[0.5cm]
\large $^1$: ARC CoE in Plant Energy Biology, ANU, Canberra\\[0.2cm]
\large $^2$: NICTA, Canberra\\[0.4cm]
\large $^3$: Research School of Biology, ANU, Canberra\\[0.4cm]
\large \texttt{kevin.murray@anu.edu.au} \hspace{1cm} Poster URL: \large
  \url{http://git.io/vcxYF}\\
\end{minipage}
\begin{minipage}[b]{0.4\linewidth}
  \hspace{5cm}
\includegraphics[width=20cm]{logo.png}\\
\end{minipage}

\vspace{1cm} % A bit of extra whitespace between the header and poster content

%------------------------------------------------------------------------------
\begin{multicols}{2}

%------------------------------------------------------------------------------
%	ABSTRACT
%------------------------------------------------------------------------------

\begin{abstract}
\vspace{5mm}

Modern techniques in population genomics generate unprecedented quantities of
data within which complex genetic histories reside. The scale and complexity of
these data require the development of new approaches to the analysis of genetic
data. We present the $k$-mer Weighted Inner Product, a \textit{de novo},
alignment free measure of genetic similarity between samples in a population.
\texttt{kWIP}, is an efficient tool implementing this metric that can determine
the genetic relatedness between samples without alignment or assembly. We show
\texttt{kWIP} can reconstruct the true relatedness between samples directly
from sequencing reads generated with various modern sequencing platforms, as
well as from simulated data.

\end{abstract}

%------------------------------------------------------------------------------
%	INTRODUCTION
%------------------------------------------------------------------------------

\section*{Introduction}

Modern population genomics requires sequencing many thousands of samples. To
distil knowledge from this data requires analysis by sequence comparison. To
compare such datasets, algorithmic improvement is required. Alignment-free
sequence comparison is a powerful field of algorithms which overcome some
shortcomings of sequence alignment. However, few alignment free algorithms can
process raw data from modern sequencing platforms, which sequence genomes as
millions of short fragments. kWIP extends alignment-free sequence comparison
algorithms to accept sequencing data directly.

%------------------------------------------------------------------------------
%	MATERIALS AND METHODS
%------------------------------------------------------------------------------

\section*{Algorithms}

\texttt{kWIP} works by decomposing sequencing reads to short $k$-mers, hashing
these $k$-mers using a constant-memory data structure, and performing pairwise
distance calculation between these sample $k$-mer hashes. One can calculate
the inner product between hashes as a similarity measure. However, this treats
all $k$-mers as of equal importance and accuracy. Therefore, \texttt{kWIP}
applies a weight to each $k$-mer to reduce the contribution of technical noise
to the overall signal, and focus on $k$-mers which provide maximal information
about relatedness within a population.

%------------------------------------------------

\subsection*{Hashing}

Sequence reads are decomposed into $k$-mers and counted in a probabilistic
data structure (Hash). This hashing is performed using the \texttt{khmer} C++
library \cite{crusoe_khmer_2015}.

\begin{center}
  \vspace{1cm}
  \includegraphics[width=20cm]{hashing.png}
  \vspace{1cm}
\end{center}

\subsection*{Entropy vector weighting}

To calculate the weighting applied to each $k$-mer, we first calculate the
frequency of occurrence of the $k$-mer in the population. This is simply the
proportion of samples with non-zero counts of a given $k$-mer.

\begin{center}
  \vspace{1cm}
  \includegraphics[width=25cm]{freq_vector.png}
  \vspace{1cm}
\end{center}

The Shannon entropy of this frequency is used as the weights of each $k$-mer,
calculated per (\ref{eqn:shannon}).

\begin{equation}
    H = - \sum_{i} P(x_i) log_2(P(x_i)
\label{eqn:shannon}
\end{equation}

\subsection*{Inner Product Calculation}

Sample similarity is calculated pairwise between all samples as the inner
product of hashes. The inner product between two hashes alone is calculated as
(\ref{eqn:ip}). The weighted inner product calculation is calculated per
(\ref{eqn:wip}).

\begin{equation}
  \langle A, B \rangle = \sum_{i} A_i \cdot B_i
\label{eqn:ip}
\end{equation}

\begin{equation}
  \langle A, B | H \rangle = \sum_{i} A_i \cdot B_i \cdot H_i
\label{eqn:wip}
\end{equation}

\subsection*{Implementation}

\texttt{kWIP} is implemented in C++11, utilising the khmer C++ library. Weighted
and unweighted inner products have been implemented. \texttt{kWIP} uses OpenMP
to parallelise distance matrix calculation in a thread-safe manner.


%------------------------------------------------------------------------------
%	RESULTS
%------------------------------------------------------------------------------

\section*{Experimental Validation}

We present an initial experimental validation of \texttt{kWIP} and show
increased performance of the weighted inner product metric compared to the
unweighted metric.

\subsection*{Methods}

Simulated population genome sequencing


\begin{center}
  \vspace{1cm}
  \includegraphics[width=25cm]{true-dendro.png}
\end{center}

\subsection*{Simulation Results}

\begin{center}
  \includegraphics[width=30cm]{mats.png}
  \vspace{1cm}
\end{center}

Above we see \texttt{kWIP}'s estimation of the genetic relatedness between
simulated samples. Note the increased resolution and fidelity of the WIP when
compared to the IP distance matrix.

\texttt{kWIP} can calculate pairwise relatedness between 96 samples of 3Gbp in
approximately 10 hours on 16 CPUs.

%------------------------------------------------------------------------------
%	CONCLUSIONS
%------------------------------------------------------------------------------

%\color{ForestGreen}


\section*{Summary}
%\Large
\texttt{kWIP}: a fast tool to determine approximate genetic relatedness
\begin{itemize}
  \item Entirely \textit{de novo} and alignment free
  \item Hashes $k$-mers into probabilistic data structures
  \item Calculates inner products between sample hashes
  \item Available from \url{https://github.com/kdmurray91/kwip} under the GNU GPL v3+.
\end{itemize}

%\normalsize
%\color{Black}

%------------------------------------------------------------------------------
%	FORTHCOMING RESEARCH
%------------------------------------------------------------------------------

\section*{Forthcoming Research}

A paper describing \texttt{kWIP} in more detail is in preparation. We plan to
deploy \texttt{kWIP} for use as a quality assurance measure across several
large-scale plant population genome sequencing projects. We plan to implement
MPI-based parallelism, and to extend kwip to full counting Bloom filters or
counting de Bruijn graphs.

%------------------------------------------------------------------------------
%	ACKNOWLEDGEMENTS
%------------------------------------------------------------------------------

\section*{Acknowledgements}

We thank Conrad Burden, Terry Neeman, Ben Kahler, Gavin Huttley and Cameron
Jack for comments or advice on algorithms and experiments presented here.

\end{multicols}
\begin{multicols}{2}


%------------------------------------------------------------------------------
%	REFERENCES
%------------------------------------------------------------------------------
\tiny
\printbibliography
\normalsize

%------------------------------------------------------------------------------
\end{multicols}
\end{document}
